exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverStatsExporter.java

```diff
@@ -169,4 +171,25 @@ static void unsafeResetExporter() {
       StackdriverStatsExporter.exporter = null;
     }
   }
+
+  /**
+   * Sets {@link MonitoredResource} for this {@link StackdriverStatsExporter}.
+   *
+   * <p>If not set, this exporter will use a default {@code MonitoredResource} with type global and
+   * no resource labels. In this case, it will be error-prone to have more than one {@code
+   * StackdriverStatsExporter}s exporting stats for the same set of {@code View}s.
+   *
+   * <p>If you have multiple {@code StackdriverStatsExporter} running at the same time, please make
+   * sure each {@code StackdriverStatsExporter} has a unique {@code MonitoredResource}.
+   *
+   * <p>If {@link #setMonitoredResource(MonitoredResource)} is called after {@code
+   * StackdriverStatsExporter} is created, all the previous stats that were uploaded won't be
+   * affected, all the future stats will be associated with the given {@code MonitoredResource}.
+   *
+   * @param monitoredResource the {@code MonitoredResource} used by this exporter.
+   * @throws NullPointerException if the given {@code MonitoredResource} is null.
+   */
+  public static void setMonitoredResource(MonitoredResource monitoredResource) {
```

Comment:
Can this API be part of the creating the exporter?

Reply:
Maybe we should switch to a builder-like API for creating the StackdriverStatsExporter.

Reply:
That'a another alternative. My concern is we already have a lot of parameters in the constructor and several factory methods, adding one more parameter and factory method seems too much. Also, Stackdriver client allows users to change MonitoredResource at run time, I think it's better to support that as well.

Reply:
Updated to have MonitoredResource in the factory methods instead. I'll create an issue about changing to builder-like API for exporter, since there are too many factory methods now.